pycbc/workflow/jobsetup.py

```diff
@@ -1166,19 +1356,92 @@ class LalappsInspinjExecutable(Executable):
     """
     The class used to create jobs for the lalapps_inspinj Executable.
     """
-    current_retention_level = Executable.FINAL_RESULT
-    def create_node(self, segment):
+    current_retention_level = Executable.CRITICAL
+    def __init__(self, cp, exe_name, universe=None, ifos=None, out_dir=None,
+                 tags=[]):
+        Executable.__init__(self, cp, exe_name, universe, ifos, out_dir,
+                            tags=tags)
+        self.cp = cp
+        self.tags = tags
+
+    def create_node(self, segment, exttrig_file=None, tags=[]):
         node = Node(self)
 
-        if self.get_opt('write-compress') is not None:
-            ext = '.xml.gz'
+        curr_tags = self.tags + tags
+        # This allows the desired number of injections to be given explicitly
+        # in the config file. Used for coh_PTF as segment length is unknown
+        # before run time.
+        if 'coh_PTF_inspiral' in self.cp.get('executables', 'inspiral'):
```

Comment:
I don't think this check should be for coh_PTF_inspiral in the inspiral section. We could run coh_PTF_inspiral (or it's successor) in an all-sky mode in the future. Why not have this test on the exttrig-file option (to tell it we are doing GRBs), or possibly a second --grb key-word argument.